pkg/report/profit_report.go

```diff
@@ -150,46 +119,34 @@ func (r *AccumulatedProfitReport) Output(symbol string) {
 		titles := []string{
 			"#",
 			"Symbol",
-			"accumulatedProfit",
-			"accumulatedProfitMA",
-			fmt.Sprintf("%dd profit", r.AccumulatedDailyProfitWindow),
+			"Total Net Profit",
+			fmt.Sprintf("Total Net Profit %sMA%d", r.Interval, r.ProfitMAWindow),
+			fmt.Sprintf("%s%d Net Profit", r.Interval, r.ShortTermProfitWindow),
 			"accumulatedFee",
-			"accumulatedNetProfit",
 			"winRatio",
 			"profitFactor",
-			"60D trades",
+			fmt.Sprintf("%s%d Trades", r.Interval, r.Window),
 		}
-		for i := 0; i < len(r.extraValues); i++ {
-			titles = append(titles, r.extraValues[i][0])
+		for i := 0; i < len(r.strategyParameters); i++ {
+			titles = append(titles, r.strategyParameters[i][0])
 		}
 		_ = tsvwiter.Write(titles)
 
 		// Output data row
-		for i := 0; i <= r.NumberOfInterval-1; i++ {
-			accumulatedProfit := r.accumulatedProfitPerDay.Index(r.IntervalWindow * i)
-			accumulatedProfitStr := fmt.Sprintf("%f", accumulatedProfit)
-			accumulatedProfitMA := r.accumulatedProfitMAPerDay.Index(r.IntervalWindow * i)
-			accumulatedProfitMAStr := fmt.Sprintf("%f", accumulatedProfitMA)
-			intervalAccumulatedProfit := r.dailyProfit.Tail(r.AccumulatedDailyProfitWindow+r.IntervalWindow*i).Sum() - r.dailyProfit.Tail(r.IntervalWindow*i).Sum()
-			intervalAccumulatedProfitStr := fmt.Sprintf("%f", intervalAccumulatedProfit)
-			accumulatedFee := fmt.Sprintf("%f", r.accumulatedFeePerDay.Index(r.IntervalWindow*i))
-			accumulatedNetProfit := fmt.Sprintf("%f", accumulatedProfit-r.accumulatedFeePerDay.Index(r.IntervalWindow*i))
-			winRatio := fmt.Sprintf("%f", r.winRatioPerDay.Index(r.IntervalWindow*i))
-			profitFactor := fmt.Sprintf("%f", r.profitFactorPerDay.Index(r.IntervalWindow*i))
-			trades := r.dailyTrades.Tail(60+r.IntervalWindow*i).Sum() - r.dailyTrades.Tail(r.IntervalWindow*i).Sum()
-			tradesStr := fmt.Sprintf("%f", trades)
+		for i := 0; i <= r.Window-1; i++ {
 			values := []string{
 				fmt.Sprintf("%d", i+1),
-				symbol, accumulatedProfitStr,
-				accumulatedProfitMAStr,
-				intervalAccumulatedProfitStr,
-				accumulatedFee,
-				accumulatedNetProfit,
-				winRatio, profitFactor,
-				tradesStr,
+				r.symbol,
+				fmt.Sprintf("%f", r.accumulatedProfitPerInterval.Last(i)),
+				fmt.Sprintf("%f", r.profitMAPerInterval.Last(i)),
+				fmt.Sprintf("%f", r.accumulatedProfitPerInterval.Last(i)-r.accumulatedProfitPerInterval.Last(i+r.ShortTermProfitWindow)),
+				fmt.Sprintf("%f", r.accumulatedFeePerInterval.Last(i)),
+				fmt.Sprintf("%f", r.winRatioPerInterval.Last(i)),
+				fmt.Sprintf("%f", r.profitFactorPerInterval.Last(i)),
+				fmt.Sprintf("%f", r.accumulatedTradesPerInterval.Last(i)),
 			}
-			for j := 0; j < len(r.extraValues); j++ {
-				values = append(values, r.extraValues[j][1])
+			for j := 0; j < len(r.strategyParameters); j++ {
+				values = append(values, r.strategyParameters[j][1])
```

Comment:
Actually you should use the CsvHeader, CsvRecords interface to output this kind of data

Reply:
done